Sources/Brave/WebFilters/AdblockResourceDownloader.swift

```diff
@@ -71,36 +85,76 @@ public actor AdblockResourceDownloader: Sendable {
   
   /// Load cached data for the given resource. Ensures this is done on the MainActor
   private func loadCachedData(for resource: ResourceDownloader.Resource) async {
-    if let fileURL = ResourceDownloader.downloadedFileURL(for: resource) {
-      let date = try? ResourceDownloader.creationDate(for: resource)
-      await handle(downloadedFileURL: fileURL, for: resource, date: date)
+    do {
+      if let downloadResult = try ResourceDownloaderStream.downloadResult(for: resource) {
+        await handle(downloadResult: downloadResult, for: resource)
+      } else {
+        switch resource {
+        case .genericContentBlockingBehaviors:
+          do {
+            // This is a special case which we load a bundled file if the downloaded file is not present
+            try await ContentBlockerManager.shared.compileBundledRuleList(for: .blockAds)
+          } catch {
+            assertionFailure("A bundled file should not fail to compile")
+            ContentBlockerManager.log.error("Failed to compile bundled content blocker resource: \(error.localizedDescription)")
```

Comment:
assertion failure is for debug only...I want it to crash in debug because the only way it will fail is if we messed up the bundled file and it should be fixed before going to production. In production this will definitely not crash and the log is there mostly in case someone changes the file without building or something pretty rare.